### DISCUSSION TITLE
Is there way to cross compile to Windows 7 ?

### DISCUSSION THREAD

**Discussion by heejit** [cogentcore/core#1299] [General]
I am getting following error I have to use go 1.20 that is the last supported golang for windows 7 ->
```
../../../go/pkg/mod/cogentcore.org/core@v0.3.5/types/types.go:10:2: package cmp is not in GOROOT (/home/jiten/sdk/go1.20/src/cmp)
note: imported by a module that requires go 1.22
main.go:24:2: package goapps/ccws is not in GOROOT (/home/jiten/sdk/go1.20/src/goapps/ccws)
main.go:23:2: package goapps/common is not in GOROOT (/home/jiten/sdk/go1.20/src/goapps/common)
../../../go/pkg/mod/cogentcore.org/core@v0.3.5/base/errors/errors.go:10:2: package log/slog is not in GOROOT (/home/jiten/sdk/go1.20/src/log/slog)
note: imported by a module that requires go 1.22
../../../go/pkg/mod/cogentcore.org/core@v0.3.5/colors/colormap/colormap.go:9:2: package maps is not in GOROOT (/home/jiten/sdk/go1.20/src/maps)
note: imported by a module that requires go 1.22
../../../go/pkg/mod/cogentcore.org/core@v0.3.5/base/slicesx/slicesx.go:10:2: package slices is not in GOROOT (/home/jiten/sdk/go1.20/src/slices)
```

**Comment by kkoreilly** [comment]
Unfortunately Go, Microsoft, Chrome, and others have all ended support for Windows 7, and we require Go 1.22 for several important features. How important is Windows 7 support for you?

**Reply by heejit** [reply]
I am planning to rewrite a app(written in FreePascal/Lazarus) in Cogent Core and currently this app running in few Windows 7 computers that is why. 

In our office setup we don't upgrade to new operating system until it is required because there are many computers we have which can not run the latest operating systems. 

I can use Cogent Core for new app only then.

**Reply by rcoreilly** [reply]
One idea would be to try to fork go 1.22 and back out the windows 7 incompatible changes -- maybe they are all in stdlib packages that you don't actually need?  It is possible someone has already done this?

Unfortunately there is really no way we can go back earlier than 1.22

**Reply by kkoreilly** [reply]
Yes, we need Go 1.22 for various features, and Go [broke](https://github.com/golang/go/issues/64622) support for Windows 7 in 1.21.4, so the only way this could possibly work would be to build for Windows 7 using a fork of Go with that breaking change reverted. If there is no way for you to update the computers to Windows 10, then we could see whether such a fork would be possible.

**Reply by heejit** [reply]
I think better for me is to use Cogent Core for new app only.
